d2layouts/d2grid/layout.go

```diff
@@ -469,6 +462,68 @@ func (gd *gridDiagram) getBestLayout(targetSize float64, columns bool) [][]*d2gr
 		return genLayout(gd.objects, nil)
 	}
 
+	var gap float64
+	if columns {
+		gap = float64(gd.verticalGap)
+	} else {
+		gap = float64(gd.horizontalGap)
+	}
+	getSize := func(o *d2graph.Object) float64 {
+		if columns {
+			return o.Height
+		} else {
+			return o.Width
+		}
+	}
+
+	debug := false
+	skipCount := 0
+	// quickly eliminate bad row groupings
+	startingCache := make(map[int]bool)
+	// try to find a layout with all rows within 1.2*targetSize
+	// skip options with a row that is 1.2*longer or shorter
+	// Note: we want a low threshold to explore good options within attemptLimit,
+	// but the best option may require a few rows that are far from the target size.
+	okThreshold := 1.2
+	// if we don't find a layout try 25% larger threshold
+	thresholdStep := 0.25
```

Comment:
I think the convention is to have this as a constant

Reply:
#1315